src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ExportStoreProvider.cs

```diff
@@ -0,0 +1,200 @@
+
+                        if (cert.HasPrivateKey)
+                        {
+                            if (privateCert != null)
+                            {
+                                // OpenSSL's PKCS12 accelerator (PKCS12_create) only supports one
+                                // private key.  The data structure supports more than one, but
+                                // being able to use that functionality requires a lot more code for
+                                // a low-usage scenario.
+                                throw new PlatformNotSupportedException(SR.NotSupported_Export_MultiplePrivateCerts);
+                            }
+
+                            privateCert = cert;
+                        }
+                        else
+                        {
+                            PushHandle(cert.Handle, publicCerts);
+                        }
+                    }
+                }
+
+                using (SafePkcs12Handle pkcs12 = Interop.Crypto.Pkcs12Create(
+                    password,
+                    privateCertKeyHandle,
+                    privateCertHandle,
+                    publicCerts))
+                {
+                    if (pkcs12.IsInvalid)
+                    {
+                        throw Interop.Crypto.CreateOpenSslCryptographicException();
+                    }
+
+                    return Interop.Crypto.OpenSslEncode(
+                        Interop.Crypto.GetPkcs12DerSize,
+                        Interop.Crypto.EncodePkcs12,
+                        pkcs12);
+                }
+            }
+        }
+
+        private static void PushHandle(IntPtr certPtr, SafeX509StackHandle publicCerts)
+        {
+            using (SafeX509Handle certHandle = Interop.Crypto.X509Duplicate(certPtr))
+            {
+                if (!Interop.Crypto.PushX509StackField(publicCerts, certHandle))
+                {
+                    throw Interop.Crypto.CreateOpenSslCryptographicException();
+                }
+
+                // The handle ownership has been transferred into the STACK_OF(X509).
+                certHandle.SetHandleAsInvalid();
+            }
+        }
+
+        private byte[] ExportPkcs7()
+        {
+            using (SafePkcs7Handle pkcs7 = Interop.Crypto.Pkcs7CreateSigned())
+            {
+                Interop.Crypto.CheckValidOpenSslHandle(pkcs7);
+
+                foreach (X509Certificate2 cert in _certs)
+                {
+                    if (!Interop.Crypto.Pkcs7AddCertificate(pkcs7, cert.Handle))
+                    {
+                        throw Interop.Crypto.CreateOpenSslCryptographicException();
+                    }
+                }
+
+                return Interop.Crypto.OpenSslEncode(
+                    handle => Interop.Crypto.GetPkcs7DerSize(handle),
+                    (handle, buf) => Interop.Crypto.EncodePkcs7(handle, buf),
+                    pkcs7);
+            }
```

Comment:
> It's probably fine, but I'm confused here by who owns what.

I'll add some comments.  I'm definitely going to have to investigate Pkcs7AddCertificate to find out why that's safe (it probably internally up-refs).  The reason we can dispose the PKCS7 handle is that we've already called Encode to turn it into the byte[]; so we don't need the native container object anymore.